server/item.go

```diff
@@ -76,84 +78,249 @@ func (s *RESTServer) SlotHandler(w http.ResponseWriter, r *http.Request, ps http
 	s.getblob(w, r, id, items.BlobID(bid))
 }
 
+// getblob will find the given blob, either in the cache or on
+// tape, and then send it as a response. If there is an error, it
+// will return an error response.
 func (s *RESTServer) getblob(w http.ResponseWriter, r *http.Request, id string, bid items.BlobID) {
 	key := fmt.Sprintf("%s+%04d", id, bid)
-	cacheContents, length, err := s.Cache.Get(key)
-	if err != nil {
+	firsttime := true
+retry:
+	content, err := s.findContent(key, id, bid, r.Method == "GET")
+	if err == items.ErrNoStore {
+		w.WriteHeader(503)
+		fmt.Fprintln(w, err)
+		return
+	} else if err == items.ErrDeleted {
+		w.WriteHeader(404)
```

Comment:
Do we want this to be a `410` status instead? It seems more semantically correct.

Reply:
Since this was a refactor, i was keeping the same external behavior. (A test expects this to be a 404.)